aws/table_aws_sagemaker_notebook_instance.go

```diff
@@ -0,0 +1,295 @@
+				Transform:   transform.FromField("NotebookInstanceName"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsSageMakerNotebookInstanceAka,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsSageMakerNotebookInstances(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO Put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	plugin.Logger(ctx).Trace("listAwsSageMakerNotebookInstances", "AWS_REGION", region)
+
+	// Create Session
+	svc, err := SageMakerService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	err = svc.ListNotebookInstancesPages(
+		&sagemaker.ListNotebookInstancesInput{},
+		func(page *sagemaker.ListNotebookInstancesOutput, isLast bool) bool {
+			for _, notebookInstance := range page.NotebookInstances {
+				d.StreamListItem(ctx, notebookInstance)
+			}
+			return !isLast
+		},
+	)
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsSageMakerNotebookInstance(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	// TODO Put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	var name string
+	if h.Item != nil {
+		name = instanceName(h.Item)
+	} else {
+		name = d.KeyColumnQuals["notebook_instance_name"].GetStringValue()
+	}
+
+	// Create service
+	svc, err := SageMakerService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &sagemaker.DescribeNotebookInstanceInput{
+		NotebookInstanceName: aws.String(name),
+	}
+
+	// Get call
+	data, err := svc.DescribeNotebookInstance(params)
+	if err != nil {
+		plugin.Logger(ctx).Debug("getAwsSagemakerNotebookInstance", "ERROR", err)
+		return nil, err
+	}
+	return data, nil
+}
+
+func listAwsSageMakerNotebookInstanceTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("listAwsSageMakerNotebookInstanceTags")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	resourceArn := h.Item.(*sagemaker.NotebookInstanceSummary).NotebookInstanceArn
+
+	// Create Session
+	svc, err := SageMakerService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// Build the params
+	params := &sagemaker.ListTagsInput{
+		ResourceArn: resourceArn,
+	}
+
+	// Get call
+	op, err := svc.ListTags(params)
+	if err != nil {
+		logger.Debug("listAwsSageMakerNotebookInstanceTags", "ERROR", err)
+		return nil, err
+	}
+
+	return op, nil
+}
+
```

Comment:
you are already getting the arn of the notebook instance.  why is this function required?